res/controllers/Traktor-Kontrol-S4-MK3.js

```diff
@@ -0,0 +1,70 @@
+            engine.setParameter("[Channel2]", "pregain", view[9]/4096);
+            engine.setParameter("[Channel4]", "pregain", view[10]/4096);
+
+            // These control the controller's audio interface in hardware, so do not map them.
+            // engine.setParameter('[Master]', 'gain', view[11]/4096);
+            // engine.setParameter('[Master]', 'booth_gain', view[12]/4096);
+            // engine.setParameter('[Master]', 'headMix', view[13]/4096);
+            // engine.setParameter('[Master]', 'headVolume', view[14]/4096);
+
+            engine.setParameter("[EffectRack1_EffectUnit1]", "mix", view[15]/4096);
+            engine.setParameter("[EffectRack1_EffectUnit1_Effect1]", "meta", view[16]/4096);
+            engine.setParameter("[EffectRack1_EffectUnit1_Effect2]", "meta", view[17]/4096);
+            engine.setParameter("[EffectRack1_EffectUnit1_Effect3]", "meta", view[18]/4096);
+
+            engine.setParameter("[EqualizerRack1_[Channel3]_Effect1]", "parameter3", view[19]/4096);
+            engine.setParameter("[EqualizerRack1_[Channel3]_Effect1]", "parameter2", view[20]/4096);
+            engine.setParameter("[EqualizerRack1_[Channel3]_Effect1]", "parameter1", view[21]/4096);
+
+            engine.setParameter("[EqualizerRack1_[Channel1]_Effect1]", "parameter3", view[22]/4096);
+            engine.setParameter("[EqualizerRack1_[Channel1]_Effect1]", "parameter2", view[23]/4096);
+            engine.setParameter("[EqualizerRack1_[Channel1]_Effect1]", "parameter1", view[24]/4096);
+
+            engine.setParameter("[EqualizerRack1_[Channel2]_Effect1]", "parameter3", view[25]/4096);
+            engine.setParameter("[EqualizerRack1_[Channel2]_Effect1]", "parameter2", view[26]/4096);
+            engine.setParameter("[EqualizerRack1_[Channel2]_Effect1]", "parameter1", view[27]/4096);
+
+            engine.setParameter("[EqualizerRack1_[Channel4]_Effect1]", "parameter3", view[28]/4096);
+            engine.setParameter("[EqualizerRack1_[Channel4]_Effect1]", "parameter2", view[29]/4096);
+            engine.setParameter("[EqualizerRack1_[Channel4]_Effect1]", "parameter1", view[30]/4096);
+
+            engine.setParameter("[QuickEffectRack1_[Channel3]]", "super1", view[31]/4096);
+            engine.setParameter("[QuickEffectRack1_[Channel1]]", "super1", view[32]/4096);
+            engine.setParameter("[QuickEffectRack1_[Channel2]]", "super1", view[33]/4096);
+            engine.setParameter("[QuickEffectRack1_[Channel4]]", "super1", view[34]/4096);
+
+            engine.setParameter("[EffectRack1_EffectUnit2]", "mix", view[35]/4096);
+            engine.setParameter("[EffectRack1_EffectUnit2_Effect1]", "meta", view[36]/4096);
+            engine.setParameter("[EffectRack1_EffectUnit2_Effect2]", "meta", view[37]/4096);
+            engine.setParameter("[EffectRack1_EffectUnit2_Effect3]", "meta", view[38]/4096);
+        }
+    }
+    init() {
+    }
+    shutdown() {
+    }
+}
+
+var TraktorS4MK3 = new S4MK3();
```

Comment:
Note that the legacy controller system fails to access `TraktorS4MK3` if `let` is used instead of `var` here :see_no_evil:

Reply:
We should consider switching to the module approach we discussed on zulip a long time ago... eg scripts export their controller as ES6 modules

Reply:
This is good enough for 2.4.